crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -117,9 +120,26 @@ impl ShaderCache {
         let module = match data.processed_shaders.entry(shader_defs.to_vec()) {
             Entry::Occupied(entry) => entry.into_mut(),
             Entry::Vacant(entry) => {
+                let mut shader_defs = shader_defs.to_vec();
```

Comment:
Curious what @superdump thinks about this one. I think I'm on board for having a suite of "low level feature detection shader defs" that exist everywhere, but this is a big enough change to merit a discussion of its own.

Reply:
yeah... I fixed a similar issue in #4949 and wanted to try a way that wouldn't need to fix it for other pipelines one by one

Reply:
I think centralising these kinds of ‘global’ shader defs based on detected features/limits makes a lot of sense. They aren’t different in different renderers or pipelines and they are useful to be able to use without core changes / user code reimplementation, in my opinion. And while specialisation has different roots, the shader cache is centralised. Good idea. This is my initial 0730 after a night out take. :)

Reply:
best kind of takes!